cfgov/jinja2/v1/_layouts/base.html

```diff
@@ -255,56 +255,46 @@
-      {% macro template_js() %}
-         {% include 'js/routes/' + request.path.split('/')[1] + '/single.js' ignore missing %}
-      {% endmacro %}
-      {% set js_source = template_js() | trim %}
-      {% if js_source | length > 0 %}
-        loadScript('{{ static('js/routes/' + request.path.split('/')[1] + '/single.js') }}');
-      {% endif %}
-    }
+    {% macro template_js() %}
+       {% include 'js/routes/' + request.path.split('/')[1] + '/single.js' ignore missing %}
+    {% endmacro %}
+    {% set js_source = template_js() | trim %}
+    {% if js_source | length > 0 %}
+      a.push( '{{ static('js/routes/' + request.path.split('/')[1] + '/single.js') }}' );
+    {% endif %}
+
     {# Check and include component-level JavaScript. #}
-    function loadScriptComponent(){
-      {% if page and page.media %}
-         {% for js in page.media %}
-           loadScript('{{ static('js/routes/on-demand/' + js) }}');
-         {% endfor %}
-      {% endif %}
-    }
+    {% if page and page.media %}
+       {% for js in page.media %}
+         a.push( '{{ static('js/routes/on-demand/' + js) }}' );
+       {% endfor %}
+    {% endif %}
+
     {# TODO: Remove the below block when all pages
              using this template are moved to Wagtail. #}
-    function loadScriptPage(){
-      {# Check and include page-level JavaScript. #}
-      {% macro page_js() %}
-         {% include 'js/routes/' + request.path[1:] + 'index.js' ignore missing %}
-      {% endmacro %}
-      {% set js_source = page_js() | trim %}
-      {% if js_source | length > 0 %}
-        loadScript('{{ static('js/routes/' + request.path[1:] + 'index.js') }}');
-      {% endif %}
-    }
-
-    if (document.body.parentElement.className.indexOf('no-js')===-1){
-      {# Include site-wide JavaScript. #}
-      loadScript('{{ static('js/routes/common.js') }}', function(){
-        loadScriptTemplate();loadScriptComponent();loadScriptPage();
-      });
-    }
-</script>
+    {# Check and include page-level JavaScript. #}
+    {% macro page_js() %}
+       {% include 'js/routes/' + request.path[1:] + 'index.js' ignore missing %}
+    {% endmacro %}
+    {% set js_source = page_js() | trim %}
+    {% if js_source | length > 0 %}
+      a.push( '{{ static('js/routes/' + request.path[1:] + 'index.js') }}' );
+    {% endif %}
 
-<script>
-//<![CDATA[
-    var usasearch_config = { siteHandle: 'cfpb' };
-    loadScript('https://search.usa.gov/javascripts/remote.loader.js');
-//]]>
+    {# Minified dynamic JavaScript loader that injects a script tag in the head
+      of the page. It also provides a callback parameter. #}
+    var b,c,d=[],f=document.scripts[0];function g(){for(var e;d[0]&&'loaded'==d[0].readyState;)e=d.shift(),e.onreadystatechange=null,f.parentNode.insertBefore(e,f)}for(;b=a.shift();)'async'in f?(c=document.createElement('script'),c.async=!1,c.src=b,document.head.appendChild(c)):f.readyState&&(c=document.createElement('script'),d.push(c),c.onreadystatechange=g,c.src=b);
```

Comment:
`readyState` can be stripped out.

Reply:
…because it's for IE9 and below, which won't handle the `async` in this PR anyway.

Reply:
If I understand correctly (and it's very very possible I am not understanding correctly), the whole `readyState` bit is to compensate for the fact that IE9 and under don't handle `async`. Check out the comments inside the `while` loop [in the article's final recommendation](https://www.html5rocks.com/en/tutorials/speed/script-loading/#toc-aggressive-optimisation).

Reply:
To add: This PR works fine for me as-is in IE9.

Reply:
@Scotchester Try reloading a few times.

Reply:
@Scotchester is correct, the readyState code is for ie9. I'll check the ie9 errors.

Reply:
Ah yeah, after a few reloads, I got a `'webpackJsonp' is undefined` error.

Reply:
The code wasn't actually being executed in IE9; `'async' in firstScript` was returning true for IE9.  Even once that was fixed, it still wasn't guaranteeing the execution order in IE9 using readyState. 

I've removed the defer and readyState code and conditionally set the async flag, if it's supported.